File: javascript/anagram/anagram.js
var Anagram = function(text) {
  this.text = text;
};

Anagram.prototype.matches = function() {
  var args = Array.prototype.slice.call(arguments);
  var words = args.reduce(function(prevVal, elem) {
    if (Array.isArray(elem) || (typeof elem === 'string')) {
      prevVal = prevVal.concat(elem);
    }
    return prevVal;
  }, []);
  var lowerCaseText = this.text.toLowerCase();
  var sortedText = lowerCaseText.split('').sort().join('');

  return words.filter(function(elem, i, array) {
    var lowerCaseElem = elem.toLowerCase();
    var sortedElem = lowerCaseElem.split('').sort().join('');
    return sortedElem === sortedText && lowerCaseElem !== lowerCaseText;
  }, sortedText, lowerCaseText);
};

module.exports = Anagram;

File: python/anagram/anagram.py
def isanagram(a, b):
    a = a.lower()
    b = b.lower()
    if a == b:
        return False
    else:
        return "".join(sorted(a)) == "".join(sorted(b))


def detect_anagrams(string, array):
    return filter(lambda x: isanagram(x, string), array)
